sass.py

```diff
@@ -57,9 +58,26 @@ def mkdirp(path):
         raise
 
 
+def _prepare_custom_function_list(custom_function_map):
+    # (signature, function_reference)
+    custom_function_list = []
+    for func_name, func in sorted(custom_function_map.items()):
+        argspec = inspect.getargspec(func)
+        if argspec.varargs or argspec.keywords or argspec.defaults:
+            raise TypeError(
+                'Functions cannot have starargs or defaults: {0} {1}'.format(
+                    func_name, func,
+                )
+            )
+        blinged_args = ['$' + arg for arg in argspec.args]
+        signature = '{0}({1})'.format(func_name, ', '.join(blinged_args))
+        custom_function_list.append((signature.encode('UTF-8'), func))
+    return custom_function_list
```

Comment:
It would be better if there is a type for custom function.  The type could have explicit fields about sass function: its name, parameter names, and an arbitrary callable.  It also could have a method that returns its signature string.  I wonder what’s your thoughts about this idea.